go/cmd/dolt/commands/sql_print.go

```diff
@@ -276,8 +276,13 @@ func getJSONProcessFunc(sch sql.Schema) pipeline.StageFunc {
 					}
 
 					validCols++
-					colStr := sqlutil.SqlColToStr(col)
-					colStr = strings.Replace(colStr, "\"", "\\\"", -1)
+					colStr := sqlutil.SqlColToStr(ctx, col)
+
+					if _, ok := col.(sql.JSONValue); !ok {
+						// don't escape for JSONValue literals
+						colStr = strings.Replace(colStr, "\"", "\\\"", -1)
+					}
```

Comment:
Might want to break this out as a utility function that others can reuse rather than repeating strings.Replace in other places